File: over-privilege /Over-privilege Native/malicious/app/src/main/java/over/Privilege/malicious/MainActivity.java
package over.Privilege.malicious;


import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;
import com.Privilege.maliciousappnative.R;

public class MainActivity extends AppCompatActivity {

    static {
        System.loadLibrary("native-lib");
    }

    private native String getStringFromNative();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView tv = findViewById(R.id.sample_text);
        tv.setText(getStringFromNative());
    }
}

File: overuse-permissions/test-case(s)/kotlin-api-calls/wifi-api-benign/benign/app/src/main/java/com/overperm/benign/MainActivity.kt
package com.overperm.benign

import android.content.Context
import android.net.wifi.WifiManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    private lateinit var wifiStateTextView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        wifiStateTextView = findViewById(R.id.wifiStateTextView)

        val enableWifiButton = findViewById<Button>(R.id.enableWifiButton)
        enableWifiButton.setOnClickListener {
            setWifiEnabled(true)
        }

        val disableWifiButton = findViewById<Button>(R.id.disableWifiButton)
        disableWifiButton.setOnClickListener {
            setWifiEnabled(false)
        }

        updateWifiStateText()
    }

    private fun setWifiEnabled(enabled: Boolean) {
        val wifiManager = applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
        wifiManager.isWifiEnabled = enabled
        updateWifiStateText()
    }

    private fun updateWifiStateText() {
        val wifiManager = applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
        val wifiState = if (wifiManager.isWifiEnabled) "Enabled" else "Disabled"
        wifiStateTextView.text = "WiFi State: $wifiState"
    }
}
